plugins/inputs/icinga2/icinga2.go

```diff
@@ -0,0 +1,123 @@
+package icinga2
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Icinga2 struct {
+	Server   string
+	Filter   string
+	Username string
+	Password string
+}
+
+type Result struct {
+	Results []Object `json:"results"`
+}
+
+type Object struct {
+	Attrs Attribute  `json:"attrs"`
+	Name  string     `json:"name"`
+	Joins struct{}   `json:"joins"`
+	Meta  struct{}   `json:"meta"`
+	Type  ObjectType `json:"type"`
+}
+
+type Attribute struct {
+	CheckCommand string  `json:"check_command"`
+	DisplayName  string  `json:"display_name"`
+	Name         string  `json:"name"`
+	State        float32 `json:"state"`
+}
+
+type ObjectType string
+
+var sampleConfig = `
+	## Required Icinga2 server address (default: "https://localhost:5665")
+	# server = "https://localhost:5665"
+	## Required Icinga2 object type ("services" or "hosts, default "services")
+	# filter = "services"
+	## Required username used for request HTTP Basic Authentication (default: "")
+	# username = ""
+	## Required password used for HTTP Basic Authentication (default: "")
+	# password = ""
+	`
+
+func (s *Icinga2) Description() string {
+	return "Read status from Icinga2"
+}
+
+func (s *Icinga2) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Icinga2) GatherStatus(acc telegraf.Accumulator, checks []Object) {
+	for _, check := range checks {
+		record := make(map[string]interface{})
+		tags := make(map[string]string)
+
+		record["name"] = check.Attrs.Name
+		record["status"] = check.Attrs.State
+
+		tags["display_name"] = check.Attrs.DisplayName
+		tags["check_command"] = check.Attrs.CheckCommand
+
+		acc.AddFields(fmt.Sprintf("icinga2_%s_status", s.Filter), record, tags)
+	}
+}
+
+func (s *Icinga2) Gather(acc telegraf.Accumulator) error {
+
+	if s.Server == "" {
+		s.Server = "https://localhost:5665"
+	}
+
+	if s.Filter == "" {
+		s.Filter = "services"
+	}
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
```

Comment:
Make a single transport that is scoped to the Icinga2 struct, instead of created each Gather.  This will prevent connections being closed